### PR TITLE
Convert utilities to typescript

### PR DESCRIPTION
**Pull request by dannon**
Ran into this earlier tracking down #15706 and wanted to add a few tests to standardize sanitization a bit, converted to typescript along the way.  WIP, smoketest.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by dannon** [comment]
Some of the typing may be overly strict; will have to consider some of the errors here.

**Review by jmchilton — APPROVED** [review]
package.json is conflicted

File: client/src/utils/utils.ts [hunk 1]
```diff
@@ -119,90 +120,26 @@ export function textify(lst) {
...
-                    options.error(response, status);
-                }
-            },
-        });
-    }
+interface getOptions {
+    url: string;
+    data?: any;
+    success?: (data: AxiosResponse) => void;
+    error?: (error: AxiosError) => void;
```
Comment (ElectronicBlueberry) [inline client/src/utils/utils.ts:128]: Interface conflicts with jsdoc comment

Comment (dannon) [reply client/src/utils/utils.ts:128]: Thanks for catching this; I updated the docstring.  Cache is no longer supported (and wasn't used anywhere), and data was previously undocumented.

**Review by ElectronicBlueberry — APPROVED** [review]

**Pull request merged**